app/jobs/webhooks/instagram_events_job.rb

```diff
@@ -38,19 +62,116 @@ def sender_id
     @entries&.dig(0, :messaging, 0, :sender, :id)
   end
 
+  def find_channel(instagram_id)
+    # There will be chances for the instagram account to be connected to a facebook page,
+    # so we need to check for both instagram and facebook page channels
+    # priority is for instagram channel which created via instagram login
+    channel = Channel::Instagram.find_by(instagram_id: instagram_id)
+    # If not found, fallback to the facebook page channel
+    channel ||= Channel::FacebookPage.find_by(instagram_id: instagram_id)
+
+    channel
+  end
+
   def event_name(messaging)
     @event_name ||= SUPPORTED_EVENTS.find { |key| messaging.key?(key) }
   end
 
-  def message(messaging)
-    ::Instagram::MessageText.new(messaging).perform
+  def message(messaging, channel)
+    if channel.is_a?(Channel::Instagram)
+      ::Instagram::Direct::MessageText.new(messaging, channel).perform
+    else
+      ::Instagram::MessageText.new(messaging, channel).perform
+    end
```

Comment:
check if we can name the Instagram::Direct as instagram as it is going to be more prominent , 

and the the other one lets do something like messenger Instagram

Reply:
Done